WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.kt

```diff
@@ -3813,7 +3813,9 @@ class EditPostActivity : LocaleAwareActivity(), EditorFragmentActivity, EditorIm
     @Subscribe(threadMode = ThreadMode.MAIN)
     fun onPostUploaded(event: OnPostUploaded) {
         val post: PostModel? = event.post
-        if (post != null && post.id == editPostRepository.id) {
+        val editPostId = editPostRepository.getPost()?.id
```

Comment:
@irfano 

If `editPostRepository.getPost()`  returns null, then editPostId will be null and id won't be accessed, then we can safely perform the comparison post.id == editPostId without throwing a NullPointerException. From the looks of the EditPostRepository, id isn't nullable. 

Are you are asking what if the `EditPostRepository` property is not initialized itself? Am I correct here? In that cause I would expect a `UninitializedPropertyAccessException` and not an NPE. I added a check for that in [08efcdc](https://github.com/wordpress-mobile/WordPress-Android/pull/20954/commits/08efcdcc847da37754daf6551dafef301045b7d9)

Reply:
> Are you are asking what if the EditPostRepository property is not initialized itself? Am I correct here? In that cause I would expect a UninitializedPropertyAccessException and not an NPE. 

Yes, I asked that because I thought you meant the initialization of `EditPostRepository`, but you actually meant the initialization of its members, such as `post`. Still, thanks for adding an extra check for `EditPostRepository`. 👍🏻